packages/devtools/lib/src/framework/framework.dart

```diff
@@ -307,13 +324,33 @@ class Framework {
     }
   }
 
-  void showInfo({String message, String title, List<CoreElement> children}) {
-    _showMessage(message: message, title: title, children: children);
+  void showInfo(
+    String messageId, {
+    String message,
+    String title,
+    List<CoreElement> children,
+  }) {
+    _showMessage(
+      messageId: messageId,
+      message: message,
+      title: title,
+      children: children,
+    );
   }
 
-  void showWarning({String message, String title, List<CoreElement> children}) {
+  void showWarning(
+    String messageId, {
```

Comment:
Does each message need an ID? If not, `messageId` could become an optional named field.